cmd/argocd/commands/app.go

```diff
@@ -1610,8 +1669,12 @@ func waitOnApplicationStatus(acdClient apiclient.Client, appName string, timeout
 		}
 	}
 
-	if timeout != 0 {
-		time.AfterFunc(time.Duration(timeout)*time.Second, func() {
+	delete := make(chan bool, 1)
+	if opts.timeout != 0 {
+		time.AfterFunc(time.Duration(opts.timeout)*time.Second, func() {
+			if opts.watch.delete {
+				delete <- false
+			}
 			cancel()
 		})
 	}
```

Comment:
I don't think this is necessary (see main comment).

Reply:
I agree, I got a bit carried away and realised that there was already a channel I could use. :)